### PR TITLE
Dependency updates

### PR DESCRIPTION
**Pull request by manuelsc**
- Updated to Capacitor 5
- Updated to Ionic 7
- Updated to Angular 16
- Updated Typescript to 5.1
- NodeJs min version set to 16, updated readme
- Billing library updated and got rid of custom typed dependency. Supports Play Billing 5.0
- Updated highcharts to 11
- Migrated ion-toggles and ion-select to new schema
- Removed ion-infinite-scroll and replaced it with angulars infinite scroll
- Use official LocalNotification and PushNotification again now that Android 13 permissions are handled

### CODE REVIEW

File: src/app/utils/ThemeUtils.ts [hunk 1]
```diff
@@ -186,8 +185,16 @@ export default class ThemeUtils {
 		const isDarkThemed = await this.isDarkThemed()
 		this.changeStatusBarColor(color, isDarkThemed)
 		this.changeNavigationBarColor(isDarkThemed)
+		setTimeout(() => {
+			this.changeNavigationBarColor(isDarkThemed)
+		}, 250)
```
Comment (D13ce) [inline src/app/utils/ThemeUtils.ts:190]: As discussed internally, please consider using `applyColorInitially`.

File: src/app/tab-blocks/tab-blocks.page.ts [hunk 2]
```diff
@@ -16,18 +17,16 @@ import { ValidatorUtils } from '../utils/ValidatorUtils'
 export class TabBlocksPage implements OnInit {
 	public classReference = UnitconvService
 
+	dataSource: InfiniteScrollDataSource<BlockResponse>
```
Comment (D13ce) [inline src/app/tab-blocks/tab-blocks.page.ts:20]: Having many blocks on the block page and scrolling down for a bit causes the following error:
```
Error while writing to local storage (might be full): DOMException: Failed to execute 'setItem' on 'Storage': Setting the value of 'CapacitorStorage.cachemodule2_api' exceeded the quota.
```

File: src/app/utils/InfiniteScrollDataSource.ts
Comment (D13ce) [inline src/app/utils/InfiniteScrollDataSource.ts:1]: Wherever the new infinite scroll is used, scrolling up triggers a page refresh on Chrome instantly now.
Before, scrolling up only caused a page refresh when the user already scrolled up to the top already.

File: src/app/utils/ThemeUtils.ts [hunk 3]
```diff
@@ -203,40 +210,40 @@ export default class ThemeUtils {
 				else NavigationBar.setBackgroundColor({ color: '#f7f7f7' })
 			}
 		} catch (e) {
-			console.warn('error in changeNavigationBarColor', e)
+			console.warn('error setting navigation bar color', e)
 		}
 	}
 
-	private async changeStatusBarColor(color, isDarkThemed) {
+	private async changeStatusBarColor(color: string, isDarkThemed) {
```
Comment (D13ce) [inline src/app/utils/ThemeUtils.ts:217]: Toggling the theme causes the following error on Chrome:
```
ERROR Error: Uncaught (in promise): Error: "StatusBar" plugin is not implemented on web
Error: "StatusBar" plugin is not implemented on web
```

**Review by D13ce — DISMISSED** [review]
I have tested your changes via Chrome and here is one more finding I am unable to pin-point the reason for:
Search for a validator and press the "bookmark" symbol. It does not turn orange even though the validator is added properly. This works properly on my mobile phone i.e. with the old version.

ptal

File: src/app/tab-blocks/tab-blocks.page.ts (hunk 2 above)
Comment (manuelsc) [reply src/app/tab-blocks/tab-blocks.page.ts:21]: Addressed by changing how we cache api requests in localStorage

**Review by D13ce — CHANGES_REQUESTED** [review]

File: src/app/components/dashboard/dashboard.component.ts [hunk 4]
```diff
@@ -257,6 +262,7 @@ export class DashboardComponent implements OnInit {
 
 	updateSmoothingPool() {
 		try {
+			if (!this.validatorUtils.rocketpoolStats || !this.validatorUtils.rocketpoolStats.effective_rpl_staked) return
```
[the remark this answers is not in the document]
Comment (manuelsc) [reply src/app/components/dashboard/dashboard.component.ts:265]: this is just to detect whether rocketpool data is in the response

File: src/app/components/validator/validator.component.ts
[the remark this answers is not in the document]
Comment (manuelsc) [reply src/app/components/validator/validator.component.ts:1]: Great find, addressed it in the most recent commit

[the remark this answers is not in the document]
Comment (D13ce) [reply src/app/components/validator/validator.component.ts:1]: I have found another bug:
1) Go to the validator page and search for a random validator you have not subscribed yet
2) Subscribe it
3) Unsubscribe it
4) Try to subscribe it again => Does not work

**Review by D13ce — APPROVED** [review]
lgtm :tada:

**Pull request merged**